test/SkipCallsTestHook.t.sol

```diff
@@ -0,0 +1,255 @@
+                            & ~Hooks.BEFORE_REMOVE_LIQUIDITY_FLAG & ~Hooks.AFTER_REMOVE_LIQUIDITY_FLAG & ~Hooks.BEFORE_SWAP_FLAG
+                            & ~Hooks.AFTER_SWAP_FLAG & ~Hooks.BEFORE_DONATE_FLAG & ~Hooks.AFTER_DONATE_FLAG
+                    )
+            )
+        );
+
+        deploy(skipCallsTestHook);
+
+        assertEq(skipCallsTestHook.counter(), 1);
+    }
+
+    function test_afterInitialize_skipIfCalledByHook() public {
+        SkipCallsTestHook skipCallsTestHook = SkipCallsTestHook(
+            address(
+                uint160(0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF)
+                    & uint160(
+                        ~Hooks.BEFORE_INITIALIZE_FLAG & ~Hooks.BEFORE_ADD_LIQUIDITY_FLAG & ~Hooks.AFTER_ADD_LIQUIDITY_FLAG
+                            & ~Hooks.BEFORE_REMOVE_LIQUIDITY_FLAG & ~Hooks.AFTER_REMOVE_LIQUIDITY_FLAG & ~Hooks.BEFORE_SWAP_FLAG
+                            & ~Hooks.AFTER_SWAP_FLAG & ~Hooks.BEFORE_DONATE_FLAG & ~Hooks.AFTER_DONATE_FLAG
+                    )
+            )
+        );
+
+        deploy(skipCallsTestHook);
+
+        assertEq(skipCallsTestHook.counter(), 1);
+    }
+
+    function test_beforeAddLiquidity_skipIfCalledByHook() public {
+        SkipCallsTestHook skipCallsTestHook = SkipCallsTestHook(
+            address(
+                uint160(0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF)
+                    & uint160(
+                        ~Hooks.BEFORE_INITIALIZE_FLAG & ~Hooks.AFTER_INITIALIZE_FLAG & ~Hooks.AFTER_ADD_LIQUIDITY_FLAG
+                            & ~Hooks.BEFORE_REMOVE_LIQUIDITY_FLAG & ~Hooks.AFTER_REMOVE_LIQUIDITY_FLAG & ~Hooks.BEFORE_SWAP_FLAG
+                            & ~Hooks.AFTER_SWAP_FLAG & ~Hooks.BEFORE_DONATE_FLAG & ~Hooks.AFTER_DONATE_FLAG
+                    )
+            )
+        );
+
+        deploy(skipCallsTestHook);
+
+        approveAndAddLiquidity(skipCallsTestHook);
+        assertEq(skipCallsTestHook.counter(), 1);
+        modifyLiquidityRouter.modifyLiquidity(key, LIQ_PARAMS, abi.encode(address(this)));
+        assertEq(skipCallsTestHook.counter(), 2);
+    }
+
+    function test_afterAddLiquidity_skipIfCalledByHook() public {
+        SkipCallsTestHook skipCallsTestHook = SkipCallsTestHook(
+            address(
+                uint160(0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF)
+                    & uint160(
+                        ~Hooks.BEFORE_INITIALIZE_FLAG & ~Hooks.AFTER_INITIALIZE_FLAG & ~Hooks.BEFORE_ADD_LIQUIDITY_FLAG
+                            & ~Hooks.BEFORE_REMOVE_LIQUIDITY_FLAG & ~Hooks.AFTER_REMOVE_LIQUIDITY_FLAG & ~Hooks.BEFORE_SWAP_FLAG
+                            & ~Hooks.AFTER_SWAP_FLAG & ~Hooks.BEFORE_DONATE_FLAG & ~Hooks.AFTER_DONATE_FLAG
+                    )
+            )
+        );
+
+        deploy(skipCallsTestHook);
+
+        approveAndAddLiquidity(skipCallsTestHook);
+        assertEq(skipCallsTestHook.counter(), 1);
```

Comment:
again comments - this surprised me because i thought it would be 1 after `deploy` so 2 after adding liquidity - but its because theres no initialize hook at all.
Might be nice to add
```
        assertEq(skipCallsTestHook.counter(), 0);
```
before adding liq too

Reply:
yeah i think theres still a couple more spots the assert counter is 0 would be nice to have